packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,138 @@
+    readonly modalDescriptionLocator: Locator;
+
+    constructor (page: Page) {
+        super(page);
+        this.componentLocator = page.getByTestId(cookieBanner.selectors.container.dataTestId);
+        this.descriptionLocator = page.getByTestId(cookieBanner.selectors.description.dataTestId);
+        this.acceptAllButtonLocator = page.getByTestId(cookieBanner.selectors.acceptAllButton.dataTestId);
+        this.necessaryOnlyButtonLocator = page.getByTestId(cookieBanner.selectors.necessaryOnlyButton.dataTestId);
+        this.managePrefsLinkLocator = page.getByTestId(cookieBanner.selectors.managePreferencesLink.dataTestId);
+        this.bodyAcceptAllLinkLocator = page.getByTestId(cookieBanner.selectors.bodyAcceptAllLink.dataTestId);
+        this.bodyNecessaryOnlyLinkLocator = page.getByTestId(cookieBanner.selectors.bodyNecessaryOnlyLink.dataTestId);
+        this.bodyManagePreferencesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyManagePreferencesLink.dataTestId);
+        this.bodyCookieStatementLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieStatementLink.dataTestId);
+        this.bodyCookieTechnologiesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieTechnologiesLink.dataTestId);
+        this.modalDescriptionLocator = page.getByTestId(cookieBanner.selectors.modalDescription.dataTestId);
+    }
+
+    async isCookieBannerVisible () {
+        return this.componentLocator.isVisible();
+    }
+
+    async clickAcceptAllButton () {
+        await this.acceptAllButtonLocator.click();
+    }
+
+    async clickBodyAcceptAllLink () {
+        await this.bodyAcceptAllLinkLocator.click();
+    }
+
+    async clickAcceptAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickAcceptAllButton() : this.clickBodyAcceptAllLink());
+    }
+
+    async getAcceptAllVariant (attribute: string) {
+        return this.acceptAllButtonLocator.getAttribute(attribute);
+    }
+
+    async clickNecessaryOnlyButton () {
+        await this.necessaryOnlyButtonLocator.click();
+    }
+
+    async getNecessaryOnlyButtonVariant (attribute: string) {
+        return this.necessaryOnlyButtonLocator.getAttribute(attribute);
+    }
+
+    async clickNecessaryOnlyLink () {
+        await this.bodyNecessaryOnlyLinkLocator.click();
+    }
+
+    async clickNecessaryOnlyAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickNecessaryOnlyButton() : this.clickNecessaryOnlyLink());
+    }
+
+    async clickManagePreferencesAction () {
+        await this.managePrefsLinkLocator.click();
+    }
+
+    async clickManagePreferencesLink () {
+        await this.bodyManagePreferencesLinkLocator.click();
+    }
+
+    async clickManagePreferencesAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickManagePreferencesAction() : this.clickManagePreferencesLink());
+    }
+
+    async clickPreferenceToggle (preferenceIds: PreferenceIds) {
```

Comment:
Small question, what is the stacktrace like in the tests if any of these page object methods fail? Do we need to handle errors to provide useful messaging?

Reply:
The stack trace comes through just like it would if the native function was run in the test. So for example, if it was trying to find a locator and couldn't find it, the stack trace would indicate that you are trying to invoke this function that calls on x particular locator and state that it cannot find it. 
For this reason, I don't think we need to add error handling.